### DISCUSSION TITLE
Grove - Laser PM2.5 Sensor (HM3301) | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#291] [Q&A]
# Grove - Laser PM2.5 Sensor (HM3301) | Seeed Studio Wiki

Grove - Laser PM2.5 Sensor (HM3301)

https://wiki.seeedstudio.com/Grove-Laser_PM2.5_Sensor-HM3301/

**Comment by rithu3** [comment]
Hi, I'm using this sensor with raspberry pi 3 model A+ for a project, but having OSError: [Errno 121] Remote I/O error.
1. Check wiring- ok
2. Verify I2C address - all ok
3. Check I2C device availability - all ok
4. Verify I2C module loading - all ok
5. Ensure proper permissions - all ok
 What else could be the problem? Raspberry pi is also working fine with other sensors.

**Reply by MatthewJeffson** [reply]
Hi, it seems that some are missing in "tech support and product discussion" section. We have offered our users several communication channels to cater to different preferences and needs.
I am sorry that I can't help you more. But I will be really grateful if you solve it out and share you projects with us! :D
Thank you in advance!

<img width="894" alt="image" src="https://github.com/Seeed-Studio/wiki-documents/assets/104187183/52c2d022-ab49-41ff-ac97-bcd920890a0d">

**Comment by Suasua44110** [comment]
Hi guys !

I need to use this sensor connected to I2C port of GrovePi, with other environmental sensors (air quality, temp&hum, dust, tVOC & CO2eq) on a Raspberry Pi 4B.

Because I’m working on Python 3.7.3, I use commands like pip3 and python3 to execute my scripts.

That’s why this tutorial does not work like that and especially, I got an error about SMBusWrapper.

What could you advise me to do ?

Thanks in advance !!!
Cheers
Sua

**Reply by xzxcessarr** [reply]
Hi, thanks for feedback! From your description we cannot distinguish where have problem because the content of your python code is unknown. You would better check the sensor by the arduino and if will, check your python code

**Comment by hbswn** [comment]
The "XIAO ESP32C3 as Bluetooth server" code (with libraries) cost 50% "of program storage space" !
"Maximum is 1310720 bytes."

**Comment by jarunge** [comment]
Good Afternoon,
I purchased this HM3301 sensor thinking it had different capabilities than it does. Upon reading the manual, I realize the HM3302 is what I need. Do you sell the HM3302 or is it available anywhere?

If not, is there a way to read the particle count concentrations rather than the mass concentrations? I would think the sensor is measuring a number concentration and then calculating a mass from it, but I would rather have just the number. If the sensor is measuring it that way, I'd like to read that off the sensor, but all the drivers only provide a way to read mass concentration.

**Comment by alkrojen** [comment]
Hello
When I run the example code basic_demo.ino on my Arduino MKR WAN 1310 I get the following output:

0   
FF  0  0  0  1   
0  1  0  1  0   
1  0  1  0  1   
0  0  0  0  0   
0  0  0  0  0   
0  0   
 
sensor num: 0
PM1.0 concentration(CF=1,Standard particulate matter,unit:ug/m3): 1
PM2.5 concentration(CF=1,Standard particulate matter,unit:ug/m3): 1
PM10 concentration(CF=1,Standard particulate matter,unit:ug/m3): 1
PM1.0 concentration(Atmospheric environment,unit:ug/m3): 1
PM2.5 concentration(Atmospheric environment,unit:ug/m3): 1
PM10 concentration(Atmospheric environment,unit:ug/m3): 1

All measurements are 1. Can this really be true that my room is that clean, or what could be wrong?

br.
Allan

**Reply by wallacebrf** [reply]
i just built mine today and i am getting between 0 and 1 for all readings. 

to test the unit, i took some tissue paper, lit an end on fire for a few seconds and blew out the flames so i have some smoke. wafted the smoke around the sensor and saw the values go into the hundreds etc and each value was not identical, each had its own unique value so based on this i ASSUME it is working.
